pennylane/templates/embeddings/angle.py

```diff
@@ -66,26 +40,41 @@ def AngleEmbedding(features, wires, rotation="X"):
     ``features`` than rotations, the circuit does not apply the remaining rotation gates.
 
     Args:
-        features (array): input array of shape ``(N,)``, where N is the number of input features to embed,
+        features (tensor-like): input tensor of shape ``(N,)``, where N is the number of input features to embed,
             with :math:`N\leq n`
-        wires (Iterable or Wires): Wires that the template acts on. Accepts an iterable of numbers or strings, or
-            a Wires object.
+        wires (Iterable): Wires that the template acts on.
         rotation (str): type of rotations used
-
     """
 
-    wires = Wires(wires)
-    n_features = _preprocess(features, wires)
-    wires = wires[:n_features]
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, features, wires, rotation="X", do_queue=True):
+
+        if rotation not in ROT:
+            raise ValueError(f"Rotation option {rotation} not recognized.")
+        self.rotation = ROT[rotation]
+
+        shape = qml.math.shape(features)
+        if len(shape) != 1:
+            raise ValueError(f"Features must be a one-dimensional tensor; got shape {shape}.")
+        n_features = shape[0]
+        if n_features > len(wires):
+            raise ValueError(
+                f"Features must be of length {len(wires)} or less; got length {n_features}."
+            )
```

Comment:
I decided to put small preprocessing code blocks into the init method to avoid an additional function call. Is that a good idea? 

I don't think we need a strict convention for this...